### PR TITLE
DOC/CI: Prepare for v0.24 release

### LINKED ISSUE #2429: Planning for next release
@raphaelquast would like the feature he added in a new release: https://github.com/SciTools/cartopy/pull/2362#issuecomment-2316096926

We haven't been getting a lot of new PRs/updates lately, so I'm wondering if anyone has any specific items they want to get in or looked at before our next release. We haven't had a formal release timeframe as far as I know, so I'd be happy making a new release sometime soon since it has been a few months.

### PR DESCRIPTION
**Pull request by greglucas**
- Release notes
- Add Python 3.13 tests and builds

closes #2429

### CODE REVIEW

**Comment by greglucas** [comment]
3.13 wheels are currently blocked by https://github.com/pyproj4/pyproj/issues/1425

**Comment by rcomer** [comment]
pyproj 3.7 is out and CI is green :tada:

**Review by rcomer — APPROVED** [review]
This seems good to me.

**Comment by greglucas** [comment]
@rcomer, I just pushed another two updates:
* Added the new LambertZoneII Projection to the list of projections
* Added v0.24 release notes to the what's new page

I'm not sure why the coastline cacheing test is failing on windows now...

**Review by rcomer — APPROVED** [review]

File: docs/source/reference/projections.rst [hunk 1]
```diff
@@ -341,6 +341,22 @@ OSGB
...
+.. autoclass:: cartopy.crs.LambertZoneII
+
+.. plot::
+
+    import matplotlib.pyplot as plt
+    import cartopy.crs as ccrs
+
+    plt.figure(figsize=(3.2687, 3))
+    ax = plt.axes(projection=ccrs.LambertZoneII())
+    ax.coastlines(resolution='10m')
```
Comment (rcomer) [inline docs/source/reference/projections.rst:356]: Actually maybe this one should add `BORDERS` rather than coastlines, so you get the outline of France?

Comment (rcomer) [reply docs/source/reference/projections.rst:356]: Too late - never mind.

Comment (greglucas) [reply docs/source/reference/projections.rst:356]: Sorry, just wanted to get this out the door so continued on! I agree with this though. I think there are a few that would be good to add borders to (if not all?).

**Comment by rcomer** [comment]
The PyPI wheels only go up to py3.10 😕

**Comment by QuLogic** [comment]
That's because the matrix generation pins `cibuildwheel` to an old version; I'm trying to see if I can fix it in #2450.

**Pull request merged**